docs/agent-policy.md

```diff
@@ -0,0 +1,169 @@
+
+## Agent policy format
+
+The policy is specified in YML, and the basic layout is:
+
+```yml
+outputs:
+  outputName1:
+    type: <output type 1>
+    ...
+  outputName2:
+    type: <output type 2>
+    ...
+inputs:
+  - type: <input type 1>
+    id: <input id 1>
+    ...
+  - type: <input type 2>
+    id: <input id 2>
+    ...
+```
+
+Each output configuration is grouped under the output name, while input configurations appear in a list directly under the root `inputs` field.
+
+For the most part, input and output configurations are passed directly to the underlying components when Agent runs them. However, there are some fields with special meaning or behavior, listed in the following sections. Fields marked "removed" are only used by agent and are removed from the configuration before forwarding it to the client component. Any fields not included below are passed through unchanged.
+
+### Input fields
+
+The following fields within input configurations are handled specially by Agent:
+
+#### `type` (string, required)
+
+The type of the input. This must match the `name` or `aliases` field in component's [input specification](component-specs.md). If the given `type` is an alias, Agent replaces it with its canonical input type (the `name` field in the spec file).
+
+#### `id` (string)
+
+The ID for this input. Each input must have a unique ID, which is used in logging and event metadata. This parameter _should_ be specified, but if it isn't present it defaults to the input type (note that this will be the canonical type, which may be different than the `type` field when using an alias).
+
+#### `use_output` (string, removed)
+
+The output this input should write to. This must match one of the output names from the same policy. Defaults to `default`.
+
+#### `log_level` (string, removed)
+
+The log level for this component. This field is not passed on to the underlying component; instead, Agent implements log level filtering itself. Possible values:
+- `error`
+- `warn` / `warning`
+- `info`
+- `debug`
+- `trace`
+
+#### `policy.revision` (string, overwritten)
+
+If the overall policy has a `revision` field (inserted by Fleet to track policy changes), its value is copied into the input's `policy.revision` field. This allows individual inputs (like Endpoint) to detect policy changes more easily.
+
+
+### Output fields
+
+#### `enabled` (boolean, removed)
+
+If present, this field determines whether the output is active. Defaults to true.
+
+#### `type` (string, required)
+
+The output type. If `use_shipper` is `false`, this value must match one of the entries in the `outputs` field for its inputs' spec files. Otherwise, the `shippers` field for its inputs' spec file must include a shipper type that supports this output. See [Component Specs](component-specs.md) for more details.
+
+#### `use_shipper` (boolean, removed)
```

Comment:
Assuming this is a change to the agent policy read by the agent itself, this would also result in changes in the Fleet UI, see at least:

* https://github.com/elastic/kibana/blob/6aef8a59060449637817eee5a5bcd96fa3bf2ff6/x-pack/plugins/fleet/common/types/models/output.ts#L29
* https://github.com/elastic/kibana/blob/6aef8a59060449637817eee5a5bcd96fa3bf2ff6/x-pack/plugins/fleet/common/openapi/components/schemas/output.yaml#L40

My comment in https://github.com/elastic/kibana/issues/141508#issuecomment-1307665963 describes how Fleet currently wants to enable the shipper.

Reply:
Was this change scheduled and coordinated? This is a breaking change to how the Elastic Agent parses a policy.